GitUI/UserControls/RevisionGrid/VisibleRowRange.cs

```diff
@@ -21,7 +21,7 @@ public VisibleRowRange(int fromIndex, int toIndex)
             ToIndex = toIndex;
         }
 
-        public int Count => ToIndex - FromIndex + 1;
```

Comment:
It should be corrected at the place where VisibleRowRange is created (line 534).

```csharp
            if (toIndex >= _graphModel.Count)
            {
                toIndex = _graphModel.Count - 1; <-- here
            }

            if (toIndex >= 0)
            {
                _visibleRowRange = new VisibleRowRange(fromIndex, toIndex);

```

Reply:
I'm not sure if I agree.
Example:
FromIndex = 0
ToIndex = 1

Then the count should be 2 (row 0 and 1). From/to are effectually from/to and including. Or do I miss something?

Reply:
I don't understand what you mean. The VisibleRowRange is meant to include its border values and as you wrote for the range <0,1> the count has to be 2. While after your change the count is 1 (1-0).
The problem is in the place when the toIndex is computed as is assigned with _graphModel.Count.
_graphModel count can not be assigned to toIndex, it has to be decreased by one in order be in the range of the 0-based collection.

Reply:
Your right. I Misread the diff. I removed the '+1' and not added it. My mistake. I will put bach the +1 in VisibleRowRange. Then it is 1(to)+0(from)+1 == 2 again. Thanks!